tiledb/sm/rest/rest_client.cc

```diff
@@ -260,50 +260,150 @@ Status RestClient::post_query_submit(
   if (query->type() == QueryType::READ)
     url += "&open_at=" + std::to_string(array->timestamp());
 
-  Buffer returned_data;
-  RETURN_NOT_OK(
-      curlc.post_data(url, serialization_type_, &serialized, &returned_data));
+  // Create the callback that will process the response buffers as they
+  // are received.
+  Buffer scratch;
+  auto write_cb = std::bind(
```

Comment:
Is there any difference between this and a lambda?

Reply:
Depending on the C++ version (11 vs 14+), there are pedantic reasons to prefer one to another. In this scenario, either could be used. I originally had a lambda here, but decided to put all of the logic into `post_data_write_cb`. I would be OK with this alternative, if you prefer:
```
auto write_cb =
  [this, &scratch, query, copy_state]
  (const bool reset, void* const contents, const size_t content_nbytes) {
    this->post_data_write_cb(reset, contents, content_nbytes, &scratch, query, copy_state);
  };
```

I prefer the bind because it is less verbose.

Reply:
If they're equivalent in C++11 then go ahead with whichever one you prefer.